helix-term/src/commands/typed.rs

```diff
@@ -321,8 +322,28 @@ fn set_line_ending(
         arg if arg.starts_with("nel") => Nel,
         _ => bail!("invalid line ending"),
     };
+    let (view, doc) = current!(cx.editor);
+    doc.line_ending = line_ending;
+
+    let mut pos = 0;
+    let transaction = Transaction::change(
+        doc.text(),
+        doc.text().lines().filter_map(|line| {
+            let current_pos = pos;
+            pos += line.len_chars();
+            match line_ending::get_line_ending(&line) {
+                Some(ending) if ending != line_ending => {
+                    let start = current_pos + line_ending::rope_end_without_line_ending(&line);
```

Comment:
This is calculating the line ending yet a second time.

https://github.com/helix-editor/helix/blob/ec21de08446656b9012c39db2db3d02c7ce3dc9c/helix-core/src/line_ending.rs#L215-L217=

`let start = pos - ending.len_chars()`